[DNM] Update matter bridge to NCS 2.4.0

markus-becker-tridonic-com opened the pull request.

markus-becker-tridonic-com committed:
Update matter bridge to NCS 2.0.2

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/NordicPlayground/matter-light-bridge?pullRequest=1) <br/>All committers have signed the CLA.

markus-becker-tridonic-com commented:
The signature change to the weak functions emberAfExternalAttributeReadCallback and emberAfExternalAttributeWriteCallback is not yet included, thus always the weak functions are called.

kkasperczyk-no commented on boards/nrf52840dk_nrf52840.overlay:
```diff
@@ -3,6 +3,13 @@
  */
 
+/ {
+	chosen {
```
If the sample is going to support OTA DFU on the nRF5340 DK, it needs the below code in its overlay as well.

markus-becker-tridonic-com replied:
I have not looked at OTA at all.

kkasperczyk-no commented on src/app_task.cpp:
```diff
@@ -124,33 +146,43 @@ k_timer sFunctionTimer;
+        }
+        index++;
+    }
+    ChipLogProgress(DeviceLayer, "Failed to add dynamic endpoint: No endpoints available!");
```
```suggestion
    ChipLogError(DeviceLayer, "Failed to add dynamic endpoint: No endpoints available!");
```

markus-becker-tridonic-com replied:
Applied

kkasperczyk-no commented on prj.conf:
```diff
@@ -4,104 +4,153 @@
 CONFIG_CHIP=y
 CONFIG_CHIP_ENABLE_PAIRING_AUTOSTART=y
 CONFIG_CHIP_PROJECT_CONFIG="src/chip_project_config.h"
+# 32774 == 0x8006 (example lock-app)
```
How about 0x8000 = 32768 ? It is not allocated to any device type https://github.com/project-chip/connectedhomeip/blob/master/docs/examples/discussion/PID_allocation_for_example_apps.md and it has corresponding DAC in the examples.

markus-becker-tridonic-com replied:
done

kkasperczyk-no commented on prj.conf:
```diff
@@ -4,104 +4,153 @@
+CONFIG_THREAD_NAME=y
+CONFIG_MPU_STACK_GUARD=y
+CONFIG_RESET_ON_FATAL_ERROR=n
 # Enable Thread 1.2 features
```
Those configs are not necessary, as they are enabled by default.

markus-becker-tridonic-com replied:
removed.

kkasperczyk-no commented on prj.conf:
```diff
@@ -4,104 +4,153 @@
-CONFIG_OPENTHREAD_THREAD_STACK_SIZE=6144
-CONFIG_OPENTHREAD_NUM_MESSAGE_BUFFERS=64
 
+# Default OpenThread network settings
```
I think this configuration is not required. It used to be supported to form the default Thread network without commissioning, but the support was dropped.

markus-becker-tridonic-com replied:
removed.

kkasperczyk-no commented on src/app_task.cpp:
```diff
@@ -620,7 +658,9 @@ void AppTask::ZigbeeEventHandler(ZigbeeShell * shell, ZigbeeShell::Event_t event
 		{
 			if (!strcmp(light.GetName(), "none")) {
-				AddDeviceEndpoint(&light, &bridgedLightEndpoint, DEVICE_TYPE_LO_ON_OFF_LIGHT);
+				AddDeviceEndpoint(&light, &bridgedLightEndpoint,
```
Considering the fact this method may fail and return -1, I think the returned value should be somehow handled.

markus-becker-tridonic-com replied:
done.

kkasperczyk-no commented on child_image/multiprotocol_rpmsg/prj_no_dfu.conf:
```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
```
prj_no_dfu.conf and prj_release.conf files are not needed in the mcuboot and multiprotocol_rpmsg directories, as long, as we have only one prj.conf file in the main sample directory.

kkasperczyk-no reviewed (commented):
Thank you a lot for the contribution! I have several initial comments.

lats1980 commented:
I got lots of compile error like:

C:\Users\lats\ncs\v2.0.2\modules\lib\matter\src\app\OperationalDeviceProxy.h:200:51: error: 'const struct chip::Dnssd::CommonResolutionData' has no member named 'interfaceId'
  200 |             interfaceId = nodeData.resolutionData.interfaceId;
      |                                                   ^~~~~~~~~~~
ninja: build stopped: subcommand failed.

Is there some missing code not included in the PR?

markus-becker-tridonic-com commented:
> I got lots of compile error like:
> 
> C:\Users\lats\ncs\v2.0.2\modules\lib\matter\src\app\OperationalDeviceProxy.h:200:51: error: 'const struct chip::Dnssd::CommonResolutionData' has no member named 'interfaceId' 200 | interfaceId = nodeData.resolutionData.interfaceId; | ^~~~~~~~~~~ ninja: build stopped: subcommand failed.
> 
> Is there some missing code not included in the PR?

Yes, indeed. E.g. see https://github.com/NordicPlayground/matter-light-bridge/pull/1#issuecomment-1380528823 and I did some more changes (regen zap). Did not have the time yet to update the PR.

Since I do not have ZigBee devices I modified this sample app and added a shell for adding new devices. This makes it harder now to add the relevant changes back into this PR.

markus-becker-tridonic-com commented:
> I got lots of compile error like:
> 
> C:\Users\lats\ncs\v2.0.2\modules\lib\matter\src\app\OperationalDeviceProxy.h:200:51: error: 'const struct chip::Dnssd::CommonResolutionData' has no member named 'interfaceId' 200 | interfaceId = nodeData.resolutionData.interfaceId; | ^~~~~~~~~~~ ninja: build stopped: subcommand failed.
> 
> Is there some missing code not included in the PR?

I have updated the PR to NCS 2.4.0. Would you be able to check?

markus-becker-tridonic-com commented:
As a side-effect this might have changed the pins to the default pins for UART1 (1.00 and 1.01).

markus-becker-tridonic-com commented:
The changes were done in https://github.com/NordicPlayground/matter-light-bridge/pull/2.
